test/index.test.js

```diff
@@ -312,4 +312,51 @@ describe('Test handling request', () => {
       }
     });
   });
+
+  it('should return custom status code and headers from error', () => {
+    const event = require('./sample-requests/GET-request-aws.json');
+
+    const handler = R((data) => {
+      expect(data).to.be.deep.equal(
+        Object.assign(
+          {},
+          event.queryStringParameters,
+          {
+            path1: 'ok',
+          },
+          {
+            authorizer: undefined,
+            headers: lowercaseKeys(event.headers),
+            context: { logStreamName: '1', awsRequestId: '1' },
+          }
+        )
+      );
+
+      const error = new Error();
+      error.statusCode = 405;
+      error.headers = { 'X-API': '1.0.0' };
```

Comment:
As I understand, error's headers are not using the same request headers, right? And it's just an optional thing.

Reply:
just additional headers